src/lib/FlightTasks/tasks/ManualAltitude/FlightTaskManualAltitude.cpp

```diff
@@ -252,6 +265,27 @@ void FlightTaskManualAltitude::_respectMaxAltitude()
 	}
 }
 
+void FlightTaskManualAltitude::_respectGroundSlowdown()
+{
+	float dist_to_ground = NAN;
+
+	// if there is a valid distance to bottom or vertical distance to home
+	if (PX4_ISFINITE(_dist_to_bottom)) {
+		dist_to_ground = _dist_to_bottom;
+
+	} else if (_sub_home_position->get().valid_alt) {
+		dist_to_ground = -(_position(2) - _sub_home_position->get().z);
+	}
```

Comment:
This is really similar to https://github.com/PX4/Firmware/blob/f229444722132a2e71d92d425597faaadcb480c4/src/lib/FlightTasks/tasks/AutoMapper/FlightTaskAutoMapper.cpp#L149-L162
Which one is best? Should we use the same code for both?